README.md
About
-----
Jobskee is an open source job board with minimal configuration and relatively small footprint.

Author
------
Elinore Tenorio (elinore.tenorio@gmail.com)  
Manila, Philippines

Stacks used
-----------

* Slim Microframework
* RedBeanPHP
* Bootstrap 3 UI
* PHPMailer
* Markdown
* etc.

Requirements
------------
* PHP 5.3 and above
* MySQL
* mod_rewrite enabled

Installation
------------
1. Export sql file
2. Update admin table with your desired username and password (sha1)
3. Upload the files
4. Update config.php with your settings
5. Change file permission of /assets/images and /assets/attachments to 777
6. Check that all .htaccess files were uploaded

Default admin login info: 
Email: admin@example.com
Password: admin

Installation Notes
------------------

### Enable PHP5.3+ using .htaccess

Some old hosting providers still use PHP5.2 version, note that Jobskee will not run on this old version.

In order to use PHP5.3+, you can edit .htaccess file in the root folder and uncomment (remove the pound sign at the beginning of) this line:

﻿`AddType application/x-httpd-php53 .php`

### Importing jobskee.sql

When you download Jobskee, you will find a database file included that you need to import to a MySQL database.

Before importing however, you can edit the file to update several things:

ADMIN ACCOUNT

You can look for this line in the .sql file

﻿`INSERT INTO admin (id, email, password) VALUES
(1, 'admin@example.com', 'd033e22ae348aeb5660fc2140aec35850c4da997')`;

﻿and change it with the values you want:

﻿`INSERT INTO admin (id, email, password) VALUES
(1, 'your desired admin email address', sha1('your desired admin password'))`;

You can also customize the default values for Categories and Cities with the values you want before importing jobskee.sql to your own database.

### Setting up your Jobskee job board

After downloading Jobskee﻿ and have setup your database and correct folder permission on `assets/attachments` and `assets/images`, you can now setup your job board by opening the `config.php` file found in the root folder.

I'd like to mention some important values in the `config.php` that you need to set in order to successfully run your job board:

APP_MODE - currently defaulted to 'development'. You need to set this to 'production' when your site is in production mode as it affects several other configuration (i.e. database, debug, etc.)

APP_THEME - currently set to 'default'. This is the default theme used by Jobskee. If you would like to customize this theme, it is recommended that you copy `/views/default` to your new theme (i.e. `/views/my_theme`) and set APP_THEME to 'my_theme'. This will ensure that you can go back to the default theme, should your theme customization produce error that you cannot recover.

SMTP SETTINGS - the default SMTP settings is Gmail friendly and should work right away when you provide your correct Gmail information. For other settings, like using your own hosting's default mail host, you must configure it correctly in order for email notifications to work.

These are the recommended settings:

Using "localhost"

﻿// SMTP SETTINGS  
define('SMTP_ENABLED', true);  
define('SMTP_AUTH', false);  
define('SMTP_URL', 'localhost');  
define('SMTP_USER', 'email@example.com');  
define('SMTP_PASS', '');  
define('SMTP_PORT', 25);  
define('SMTP_SECURE', '');  

and using Gmail

// SMTP SETTINGS  
define('SMTP_ENABLED', true);  
define('SMTP_AUTH', true);  
define('SMTP_URL', 'smtp.gmail.com');  
define('SMTP_USER', 'email@gmail.com');  
define('SMTP_PASS', 'gmail password);  
define('SMTP_PORT', 465);  
define('SMTP_SECURE', 'ssl');  

APPLICATION URL PATHS - as commented in the file, you need to provide your full URL including the trailing slashes.

SHARETHIS_PUBID - in order to enable the social media sharing for the jobs, you need to register a Publication ID at www.sharethis.com

CRON_TOKEN - this is used for running cron job to expire jobs. Provide a unique token that you can use in order to expire jobs using the path: `/cron/jobs/expire/:cron_token`

GA_TRACKING - get insights to your job board by adding a Google Analytics tracking ID here.

### Switch Language

Comment out Lines 15 and 16 and uncomment Lines 19 and 20 to switch from English to French translation

CHANGELOG
==== CHANGELOG ====

Version 1.1.2 (10/10/2015)
-------------
  * #3 Fixed search bug

Version 1.1.1 (09/14/2015)
-------------
  * Added check for empty or invalid Email Address when posting a job

Version 1.1.0 (08/30/2015)
-------------
  * Added support for multi-language translation (sponsored by mytux.fr)
  * Added jobberBase to Jobskee DB migration support (see jobberbase2jobskee.php script)

Version 1.0.2 (09/21/2014)
-------------
  * Fixed missing CSRF on search 
  * Removed ShareThis and replaced with AddThis as block (Admin->Manage->Blocks)
  * Removed deprecated search
  * Added debug option (turns off when APP_MODE is 'production')
  * Added footer message on emails sent (/models/Notifications.php)
  * Changed highlighting for featured job on search results page

Version 1.0.1 (02/01/2014)
-------------
  * Added Subscription manager in admin
  * Added Slim Extra for CSRF
  * Added site referrer validation

Version 1.0.0 (01/01/2014)
-------------
  * Added Blocks module
  * Fixed 404 errors

Version 0.9 (12/14/2013)
-----------
  * Added ban IP and email address
  * Added page manager
  * Added bulk upload using CSV
  * Added categories and cities manager
  * Added Notification class to handle all notification messages

Version 0.8 (12/7/2013)
-----------
  * Separated admin templates to another folder
  * Added viewer for job applications submitted
  * Updated config file with template paths
  * Added MarkDown editor for Bootstrap (toopay.github.com/bootstrap-markdown)
  * Added email subscription

Version 0.7 (12/2/2013)
-----------
  * Removed companies module
  * Added separate admin templates
  * Added search functionality

Version 0.6 (11/28/2013)
-----------
  * Added Parsedown for MarkDown rendering (github.com/erusev/parsedown)
  * Updated forms
  * Added job expiration for cron job maintenance

Version 0.5 (11/25/2013)
-----------
  * Added job application feature

Version 0.4 (11/23/2013)
-----------
  * Added admin authentication
  * Added job city filter

Version 0.3 (11/19/2013)
-----------
  * Added pagination
  * Added page views

Version 0.2 (11/18/2013)
-----------
  * Created separate routing controllers
  * Added front page jobs listing
  * Added job posting
  * Added job category filter
  * Added SEO title and description
  * Added OG tags for FB sharing

Version 0.1 (11/14/2013)
-----------
  * Initial commit
  * Added Slim Microframework for routing (slimframework.com)
  * Added RedBeanPHP for ORM (redbeanphp.com)
  * Added Bootstrap 3 for UI (getbootstrap.com)
  * Added PHPMailer (github.com/Synchro/PHPMailer)

config.php
<?php
/**
 * Jobskee - open source job board
 *
 * @author      Elinore Tenorio <elinore.tenorio@gmail.com>
 * @license     MIT
 * @url         http://www.jobskee.com
 */

// INITIATE SESSION
session_start();
ini_set('default_charset', 'utf-8');

// ENGLISH TRANSLATION
define('APP_LANG', 'en');
setlocale(LC_ALL, 'en_EN');

// FRENCH TRANSLATION
// define('APP_LANG', 'fr');
// setlocale(LC_ALL, 'fr_FR');

// AUTO LOAD MODELS
include 'models/helpers.php';
include 'models/class.phpmailer.php';
include 'models/class.smtp.php';
include 'models/rb.php';
spl_autoload_register(function ($class) {
	if (file_exists("models/{$class}.php")) { include "models/{$class}.php"; }	
});

// LOAD TRANSLATION
$lang = new Translate();

// APPLICATION SETTINGS
define('APP_NAME', $lang->t('app|name'));
define('APP_DESC', $lang->t('app|desc'));
define('APP_AUTHOR', 'Elinore Tenorio (elinore.tenorio@gmail.com)');
define('APP_MODE', 'development'); // set to 'production' if site is live, affects RedBean not being frozen if not in correct mode
define('APP_THEME', 'default'); // set to the theme folder name you are using, found in /views directory

// TIMEZONE
date_default_timezone_set($lang->t('app|timezone'));

/*
 ****************************************************
 * USER SETTINGS
 * You are free to modify the settings below
 * to suit your needs
 ****************************************************
 */

// SMTP SETTINGS
define('SMTP_ENABLED', true);
define('SMTP_AUTH', true);
define('SMTP_URL', 'smtp.gmail.com');
define('SMTP_USER', 'username@gmail.com');
define('SMTP_PASS', 'password');
define('SMTP_PORT', 465);
define('SMTP_SECURE', 'ssl');

// APPLICATION URL PATHS
define('BASE_URL','http://jobskee.local:10088/'); // always include the trailing slash at the end

// DATABASE SETTINGS
define('DB_HOST', 'localhost');
define('DB_NAME', 'jobskee');
define('DB_USER', 'root');
define('DB_PASS', '');

// GOOGLE ANALYTICS TRACKING
define('GA_TRACKING', '');

// APP SETTINGS CONSTANTS
define('LOGO_H', 200); // logo height
define('LOGO_W', 200); // logo width
define('LIMIT', 20); // number of jobs to display per page
define('HOME_LIMIT', 5); // number of jobs to display per category on the homepage
define('EXPIRE_JOBS', 30); // days to expire jobs
define('CRON_TOKEN', 'ioYgaCEfCEXQtzP2'); // token to verify cron job execution
define('ALLOW_JOB_POST', 1); // set (1) to allow job posting and (0) to turn off

/*
 ****************************************************
 * DEFAULT CORE SETTINGS
 * Modify only if you know what you are doing
 ****************************************************
 */

// CORE APPLICATION PATH
define('APP_PATH', str_replace(DIRECTORY_SEPARATOR.DIRECTORY_SEPARATOR, DIRECTORY_SEPARATOR, dirname(__FILE__)) . DIRECTORY_SEPARATOR);
define('ADMIN_URL', BASE_URL . 'admin/'); // always include the trailing slash at the end

// CORE CONSTANTS
define('ACTIVE', 1);
define('INACTIVE', 0);
define('CSV_FIELDS', 10);

// CORE APPLICATION URLS
define('ASSET_URL', BASE_URL . 'assets/');
define('ATTACHMENT_PATH', 'assets/attachments/');
define('IMAGE_PATH', 'assets/images/');
define('LOGIN_URL', ADMIN_URL . 'login');

// MVC PATHS
define('MODEL_PATH', 'models/');
define('VIEWS_PATH', 'views');
define('CONTROLLER_PATH','controllers/');

// CORE THEME SETTINGS
define('THEME_PATH', APP_THEME .'/');
define('THEME_ASSETS', BASE_URL . VIEWS_PATH .'/'. THEME_PATH .'assets/');

// ADMIN THEME SETTINGS
define('ADMIN_THEME', 'admin/');
define('ADMIN_ASSETS', BASE_URL . VIEWS_PATH .'/'. ADMIN_THEME . 'assets/');

// REDBEAN ORM CONFIG
R::setup("mysql:host=".DB_HOST.";dbname=".DB_NAME, DB_USER, DB_PASS);
if (APP_MODE == 'production') {
    R::freeze();
	$debug = false;
} else {
	$debug = true;
}

// SLIM MICROFRAMEWORK
require 'Slim/Slim.php';
\Slim\Slim::registerAutoloader();

// SLIM CSRF GUARD
require 'Slim/Extras/Middleware/CsrfGuard.php';

$app = new \Slim\Slim(array('mode'=>APP_MODE, 'templates.path'=>VIEWS_PATH, 'debug'=>$debug));
$app->add(new \Slim\Extras\Middleware\CsrfGuard());
$app->notFound(function () use ($app) {
    $app->flash('danger', 'The page you are looking for could not be found.');
    $url = (userIsValid()) ? ADMIN_MANAGE : BASE_URL;
    $app->redirect($url);
});

$app->flashKeep();